google/cloud/internal/retry_policy.h

```diff
@@ -24,6 +24,9 @@ namespace google {
 namespace cloud {
 inline namespace GOOGLE_CLOUD_CPP_NS {
 namespace internal {
+
+enum class IsIdempotent { kIdempotent, kNonIdempotent };
```

Comment:
The "Is" part seems to re-imply the binary/booly-ness.

Might I suggest something more like `Idempotency"?